### PR TITLE
Add initial platform support docs

### LINKED ISSUE #177: Documenting Platform Support
As discussed in the last meeting, we want to clarify what platforms, vendors, and versions we support, and what we're focusing on right on now. the outcome of this would be a page in the GPU dev guide that provides an overview the support and what that kind of support means for the platforms.

### PR DESCRIPTION
**Pull request by XAMPPRocky**
### [Rendered](https://github.com/EmbarkStudios/rust-gpu/blob/platform-support/docs/src/platform-support.md)

closes #177

### CODE REVIEW

File: docs/src/platform-support.md [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+- **Primary —** Built and tested on CI.
+- **Secondary —** Built but *not fully tested* on CI.
+- **Tertiary —** Present in the codebase but not built or tested.
+
+## Operating System
+
+| Operating System | Version | Support | Notes
+|-------------------|---------|---------|-------
+| Windows | 10+ | Primary | |
+| Linux | ??? | Primary | |
```
Comment (khyperia) [inline docs/src/platform-support.md:13]: maybe the ubuntu CI version we use here?

**Review by khyperia — APPROVED** [review]

File: docs/src/platform-support.md (hunk 1 above)
Comment (repi) [reply docs/src/platform-support.md:13]: That's a good idea and could be a base

**Review by Jasper-Bekkers — APPROVED** [review]
We probably should just link to `https://vulkan.lunarg.com/doc/view/latest/windows/vulkaninfo.html` instead of a specific version of `vulkaninfo`. Otherwise this looks good.

**Comment by XAMPPRocky** [comment]
@Jasper-Bekkers Good catch, I just got the URL from a search. Updated now.

**Review by repi — APPROVED** [review]

**Pull request merged**